### PR TITLE
Task/csi 4567 finish pr 514

### PR DESCRIPTION
**Pull request by matancarmeli7**
resolving #514

### CODE REVIEW

File: controllers/servers/host_definer/watcher/watcher_helper.py [hunk 1]
```diff
@@ -210,54 +230,70 @@ def _decode_base64_to_string(self, content_with_base64):
...
-            self._create_definition(host_definition)
+            host_definition_info = self._get_host_definition_info_from_secret_and_node_name(node_name, secret_id)
+            self._create_definition(host_definition_info)
+
+    def _generate_secret_id(self, secret_name, secret_namespace):
+        return (secret_name, secret_namespace)
 
-    def _generate_secret_id_from_secret_and_namespace(self, secret_name, secret_namespace):
-        return secret_name + settings.SECRET_SEPARATOR + secret_namespace
+    def _munch_watch_event(self, watch_event):
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/watcher/watcher_helper.py:298]: `_munch` is sufficient I believe, there is no special logic for the `watch_event`..

File: controllers/servers/host_definer/kubernetes_manager/manager.py [hunk 2]
```diff
@@ -193,61 +211,63 @@ def _get_status_manifest(self, host_definition_phase):
 
         return status
 
-    def _get_event_for_host_definition(self, host_definition, message, action, message_type):
+    def _generate_k8s_event_for_host_definition(self, host_definition_info, message, action, message_type):
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/kubernetes_manager/manager.py:214]: do we have k8s_event generation for another purpose than host_definition

Comment (matancarmeli7) [reply controllers/servers/host_definer/kubernetes_manager/manager.py:214]: no

File: controllers/servers/host_definer/watcher/watcher_helper.py [hunk 3]
```diff
@@ -210,54 +230,70 @@ def _decode_base64_to_string(self, content_with_base64):
...
         for secret_id, storage_classes_using_this_secret in SECRET_IDS.items():
             if storage_classes_using_this_secret == 0:
                 continue
-            host_definition = self._get_host_definition_from_secret_and_node_name(node_name, secret_id)
-            self._create_definition(host_definition)
+            host_definition_info = self._get_host_definition_info_from_secret_and_node_name(node_name, secret_id)
+            self._create_definition(host_definition_info)
+
+    def _generate_secret_id(self, secret_name, secret_namespace):
+        return (secret_name, secret_namespace)
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/watcher/watcher_helper.py:296]: why is that a function?
what the diff here:
`return self._generate_secret_id("secret_name", "secret_namespace")`
to:
`return ("secret_name", "secret_namespace")`
?

Comment (matancarmeli7) [reply controllers/servers/host_definer/watcher/watcher_helper.py:296]: for clarifying what a secret_id should look like

File: controllers/servers/host_definer/kubernetes_manager/manager.py (hunk 2 above)
Comment (matancarmeli7) [reply controllers/servers/host_definer/kubernetes_manager/manager.py]: I changed it to `_generate_k8s_event`

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Review by ELENAGER — APPROVED** [review]
LGTM

File: controllers/tests/controller_server/host_definer_server_test.py [hunk 4]
```diff
@@ -26,14 +27,14 @@ def setUp(self):
...
-        self.request = Mock(spec_set=['prefix', 'connectivity_type', 'node_id', 'system_info'])
+        self.request = Mock(spec_set=['prefix', 'connectivity_type', 'node_id', 'array_connection_info'])
 
         self.iqn = 'iqn.1994-05.com.redhat:686358c930fe'
         self.nqn = 'nqn.2014-08.org.nvmexpress:uuid:b57708c7-5bb6-46a0-b2af-9d824bf539e1'
         self.request.prefix = None
         self.request.connectivity_type = 'fc'
         self.request.node_id = '{};;;{}'.format(HOST_NAME, self.iqn)
-        self.request.system_info = SECRET
+        self.request.array_connection_info = get_array_connection_info_from_secrets(SECRET)
```
Comment (ArbelNathan) [inline controllers/tests/controller_server/host_definer_server_test.py:37]: do we wanna use the actual function or mock it?
what if there is an "error" in the function?
@ELENAGER what do you think?

Comment (ELENAGER) [reply controllers/tests/controller_server/host_definer_server_test.py:37]: we are not testing function get_array_connection_info_from_secrets, so, I think we need to use a mock

Comment (matancarmeli7) [reply controllers/tests/controller_server/host_definer_server_test.py]: done

**Review by ELENAGER — DISMISSED** [review]

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Review by ELENAGER — APPROVED** [review]
LGTM

**Pull request merged**